airflow/jobs/scheduler_job.py

```diff
@@ -839,17 +843,17 @@ def _change_state_for_tis_without_dagrun(
         :type old_states: list[airflow.utils.state.State]
         :param new_state: set TaskInstances to this state
         :type new_state: airflow.utils.state.State
-        :param simple_dag_bag: TaskInstances associated with DAGs in the
-            simple_dag_bag and with states in the old_states will be examined
-        :type simple_dag_bag: airflow.utils.dag_processing.SimpleDagBag
+        :param dag_ids: TaskInstances associated with these DAG IDs
+            with states in the old_states will be examined
+        :type dag_ids: list[str]
         """
         tis_changed = 0
         query = session \
             .query(models.TaskInstance) \
             .outerjoin(models.DagRun, and_(
                 models.TaskInstance.dag_id == models.DagRun.dag_id,
                 models.TaskInstance.execution_date == models.DagRun.execution_date)) \
```

Comment:
We should change this to use the new relationship we've added

```suggestion
            .outerjoin(models.TaskInstance.dag_run) \
```

Reply:
Updated: https://github.com/astronomer/airflow/pull/956/commits/6dd35bcc3986429ebed15b7287fc204a7d4fdcb0